docs/enhancements/install-strategy.md

```diff
@@ -0,0 +1,196 @@
+- Implementing retries within Hive itself. Retries will be left to specific implementations of InstallStrategy interface, if desired.
+
+## Proposal
+
+Introduce a new InstallStrategy CRD interface/contract in Hive API, breaking out of the current ClusterDeployment.Spec and instead linking to one and use duck typing to interact. Initial implementation would be AgentInstallStrategy. In the future we would hopefully migrate current openshift-install pods to OpenShiftInstallStrategy, and add HypershiftInstallStrategy, etc.
+
+Hive InstallStrategy implementations are a day 1 provisioning API. They would adhere to a contract with hive for what fields/conditions they must contain and update as the installation proceeds. Hive would copy relevant data back to the ClusterDeployment as appropriate at various stages of the provisioning.
+
+ClusterDeployment becomes a central index of clusters and kubeconfigs, used for day 2 management.
+
+AgentInstallStrategy becomes a day 1 resource and will offer more flexibility for the agent controllers to define settings they need in isolation, and track their relevant status.
+
+
+### User Stories
+
+#### Story 1
+
+#### Story 2
+
+## Design Details
+
+
+### ClusterDeployment Changes
+
+ClusterDeployment should now contain only information needed to connect to the cluster and settings for day 2 management. (admin kubeconfig, admin password, ingress, control plane certs, etc) Platform section will remain for information we need, including cloud provider credentials for things like hibernation and managed DNS.
+
+ClusterDeployment.Spec.InstallStrategy goes away, and is instead replaced with InstallStrategyRef, which would look like the following:
+
+```yaml
+  installStrategyRef:
+    apiVersion: hive.openshift.io/v1
+    kind: AgentInstallStrategy
+    name: mycluster
+```
+
+InstallStrategyRef is optional, it would not be specified for an adopted cluster.
+
+Hive controllers would then interact with this strategy dynamically with duck typing, similar to Cluster API, Knative, etc.
+
+ClusterDeployment.Spec.Provisioning remains but will eventually be deprecated. Validate this is not used in conjunction with ClusterDeployment.Spec.InstallStrategyRef, must choose one path or the other.
+
+### InstallStrategy Interface
+
+
+Field | Type | Description
+----- | ---- | -----------
+spec.clusterDeploymentRef | LocalObjectReference | Link to the required ClusterDeployment for this install. Bidirectional reference. Common data should live on ClusterDeployment and never be duplicated on InstallStrategy.
+spec.imageSetRef | ClusterImageSetReference | Core mechanism in Hive for choosing what version of OpenShift to install.
+spec.stage | string | Enum string for stage of installation. (Initializing, RequirementsNotMet, Provisioning, RetryProvisioning, Complete, Failed) More details in conditions. *TODO*: Should we just work with conditions? Loss of state in disaster recovery?
```

Comment:
+1 for conditions

Reply:
RequirementsMet is probably the one we want, I'll remove Initializing.

Provisioned sounds good as well.

Reply:
So one last issue to resolve, in a DR scenario status will be lost. We need to record somewhere that this was officially completed in spec, and technically, we need to think about an install underway that needs cleanup and a retry. 

So I'm still wondering about a Stage in the Spec, even if it's just "InProgress" and "Installed" and everything else is conditions. Technically though we do have an installed flag on ClusterDeployment.Spec we're committed too, maybe this is enough. If we restored and saw ClusterDeployment.Installed = true, ClusterInstall controllers should not act on that ClusterInstall. 

I'll drop Stage for now but let me know what you think.